androidApp/src/main/java/dev/johnoreilly/confetti/MainActivity.kt

```diff
@@ -64,28 +81,50 @@ class MainActivity : ComponentActivity() {
                         navController = navController,
                         windowSizeClass = windowSizeClass,
                         displayFeatures = displayFeatures,
+                        confettiRepository = confettiRepository,
+                        defaultConferenceParameter = defaultConferenceParameter,
                     )
                 }
             }
         }
     }
+
+    /**
+     * This relies on our deep link looking either like
+     * `confetti://confetti/sessions/{conference}`
+     * or
+     * `confetti://confetti/speakers/{conference}/{speakerId}`
+     * or more generally
+     * `whatever/[sessions|speakers]/{conference}/whatever
+     *
+     * Should be doing some more general checks here or find a way to find what the original
+     * URI template was like, so we can find what is in the position of {conference} for example.
+     * Now as I see it we only get the already decoded Uri so I don't know how to get the ID in a
+     * better way.
+     */
+    private fun Uri.findConferenceParameter(): String? {
+        return pathSegments.also { Log.d("Stelios", "segmn$it") }
+            .dropWhile<String?> { segment -> segment != "sessions" && segment != "speakers" }
+            .drop(1) // Also drop the "sessions" or the "speakers" segment
+            .firstOrNull()
+    }
 }
```

Comment:
This whole section relies on the urls looking exactly like this as I am describing here. If there's a better way to do this please do let me know. Went off of our discussion here https://kotlinlang.slack.com/archives/C051P2HUVKP/p1681162362863849?thread_ts=1680994152.780399&cid=C051P2HUVKP